src/components/Login/Login.jsx

```diff
@@ -52,20 +57,23 @@ const Login = ({ setUserName }) => {
         password: userPassword,
       };
 
-      const response = await fetch("http://localhost:4000/login", {
-        method: "POST",
-        body: JSON.stringify(user),
-        headers: {
-          "Content-Type": "application/json",
-        },
-      });
-      const result = await response.json();
-      if (result.successful) {
-        localStorage.setItem("token", JSON.stringify(result));
-        setUserName(result.user.name);
-        navigate("/courses");
-      } else {
-        alert(result.message || "Login failed");
+      try {
+        const result = await loginService(user);
+        if (result.successful) {
+          localStorage.setItem(TOKEN, result.token);
```

Comment:
We can omit doing `localStorage.setItem` here and instead rely on `dispatch(setUser())` to do the work for us. I.e., we move LS work to our actions and do `localStorage.setItem` in the `setUser` action

Reply:
done